src/third-party/arduino-json-5.2/include/ArduinoJson/Arduino/String.hpp

```diff
@@ -9,11 +9,8 @@
 
 #ifndef ARDUINO
 
-#ifndef ARDUINO_STRING_OVERRIDE

```

Comment:
No, since they are just typedefing I assume they are using strings in a very basic way (probably just c_str()). Where we actually call methods on arduino's String and I built a wrapper that adapts std::string. 

See https://github.com/ed7coyne/arduino-mock/blob/0f7f9fa2cea822edf7b11099110b3e0fdb895a53/include/arduino-mock/Arduino.h#L103